File: backend/model/bill.go
package model

import (
	"fmt"
	"time"
)

type Bill struct {
	ID        int64     `db:"id" json:"id"`
	TableID   int64     `db:"table_id" json:"table_id"`
	Amount    int64     `db:"amount" json:"amount"`
	CreatedAt time.Time `db:"created_at" json:"created_at"`
}

type BillAlreadyExistError struct {
	TableID int64
}

func (e BillAlreadyExistError) Error() string {
	return fmt.Sprintf("table id %d have already paid a bill", e.TableID)
}

File: frontend/src/model/bill.ts
export interface Bill {
  id: number;
  tableId: number;
  amount: number;
  createdAt: Date;
}

export interface BillDTO {
  id: number;
  table_id: number;
  amount: number;
  created_at: string;
}

export const convertToBill = (object: BillDTO): Bill => {
  return {
    id: object.id,
    tableId: object.table_id,
    amount: object.amount,
    createdAt: new Date(object.created_at),
  }
}
